### ISSUE TITLE
Succinct publicly auditable mpc

### ISSUE THREAD

**Issue by amiller** [initc3/HoneyBadgerMPC#169]
Our toolkit lends itself to publicly auditable MPC since we are already including pedersen commitments alongside the AVSSd values, and the preprocessing values are all generated from avssd values.
Here's a question though: is there a way to do this auditing efficiently using a snark or bullet proof? 
 Reference on publicly auditable MPC: https://link.springer.com/chapter/10.1007/978-3-319-10879-7_11

**Comment by amiller** [comment]
Here's a very closely related paper:
https://eprint.iacr.org/2017/013.pdf
It is based on Trinocchio, an approach for computing SNARKs in MPC about secret shared data. The main difference here is that the snark protocol is adaptive, in the sense that you can have a commitment to the secret shared data and then later determine the circuit to evaluate on it. In either case, it makes use of commitments to secret shared data and a SNARK protocol, the result being publicly auditable MPC with a succinct proof and verification time. This looks very close to adequate for what we need, the question being can we boost it to asynchronous / linear overhead?

**Comment by amiller** [comment]
The Adaptive Trinocchio protocol indeed seems directly to address this case. The next question then is whether we can get succinct publicly auditable MPC, with either transparent CRS or universal/updateable CRS.

**Comment by Mikerah** [comment]
The answer is yes.
See : https://arxiv.org/abs/2107.04248
Oh wait. You are an author on this paper too!

**Comment by amiller** [comment]
Yep thanks @sanket1729 for establishing the proof of concept for this! Ill leave the issue open until we have an integrated demo
